base/src/main/java/org/aya/resolve/context/ModuleSymbol.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020-2023 Tesla (Yinsen) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.resolve.context;
+
+import kala.collection.Map;
+import kala.collection.MapView;
+import kala.collection.SetView;
+import kala.control.Option;
+import kala.control.Result;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+
+public interface ModuleSymbol<T extends ContextUnit> {
+
+  /**
+   * Getting the candidates of an unqualified name
+   *
+   * @param unqualifiedName the unqualified name
+   * @return the candidates, probably empty
+   */
+  @NotNull Map<ModulePath, T> getCandidates(@NotNull String unqualifiedName);
+
+  /**
+   * Getting a symbol of an unqualified name {@param unqualifiedName} in component {@param component}
+   *
+   * @param component       the component
+   * @param unqualifiedName the unqualified name
+   * @return none if not found
+   */
+  @NotNull Option<T> getQualifiedMaybe(@NotNull ModulePath component, @NotNull String unqualifiedName);
+
+  /**
+   * Trying to get a symbol of an unqualified name definitely.
+   *
+   * @param unqualifiedName the unqualified name
+   */
+  @NotNull Result<T, Error> getUnqualifiedDefinitely(@NotNull String unqualifiedName);
+
+  /**
+   * Trying to get a symbol of an optional component and an unqualified name.
+   *
+   * @param component       an optional component, none if `This`
+   * @param unqualifiedName the unqualified name
+   */
+  @NotNull Result<T, Error> getDefinitely(@NotNull ModulePath component, @NotNull String unqualifiedName);
+
+  boolean contains(@NotNull String unqualified);
+
+  boolean containsDefinitely(@NotNull String unqualified);
```

Comment:
What are the differences of these methods?

Possible to implement with `getCandidates().isNotEmpty()`?